src/main/java/org/spongepowered/common/mixin/core/server/MixinMinecraftServer.java

```diff
@@ -531,6 +530,33 @@ protected void prepareSpawnArea(WorldServer world) {
     }
 
     @Override
+    public Optional<World> copyWorld(World originalWorld, String copyName) {
+        try {
+            FileUtils.copyDirectory(getWorldDir(originalWorld.getName()), getWorldDir(copyName));
```

Comment:
Also, this wouldn't work very well for the "root" world (dimension 0) as it copies all the data, not just the current world.